include/storage/fbw.h

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2017, 2020 Nordic Semiconductor ASA
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for flash buffered writes
+ */
+
+#ifndef ZEPHYR_INCLUDE_STORAGE_FBW_H_
+#define ZEPHYR_INCLUDE_STORAGE_FBW_H_
+
+/**
+ * @brief Abstraction over flash buffered writes.
+ *
+ * @defgroup fbw flash buffered write Interface
+ * @{
+ */
+
+#include <stdbool.h>
+#include <drivers/flash.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @typedef fbw_callback_t
+ *
+ * @details The fbw callback is called when data from the buffer is written
```

Comment:
This is a little confusing to somebody unfamiliar with the API.  From the first sentence of the details it might appear this is invoked *to* write the data to flash, not *after*.  Something like this would be less ambiguous: state the purpose and use right away, not in the third sentence.

```

/**
 * @brief Signature for callback invoked after flash write completes.
 *
 * Functions of this type are invoked with a buffer containing
 * data read back from the flash after a flash write has completed.
 * Blah blah verifying the data...
 */
```

Reply:
Also: this callback must be **isr-ok** if `fbw_flash_write()` can be invoked from an interrupt.  If you want to document API behavior.

Reply:
I have not marked `fbw_flash_write` as `isr-ok` so AFAICS the callback should not have that property either.

Added your proposal to the doc.